tokio/src/runtime/task/join.rs

```diff
@@ -203,6 +203,37 @@ impl<T> JoinHandle<T> {
         }
     }
 
+    /// Checks if the task associated with the handle has finished(whether completed or cancelled).
+    /// This function does not block.
```

Comment:
The implementation seems fine, but I would like to see better documentation. It should explain that aborting a task may take some time, and that this method only returns true once the task has completed the cancellation process.

(The same applies to the `AbortHandle`)